src/Tizen.NUI.Wearable/src/public/CircularSlider.cs

```diff
@@ -0,0 +1,674 @@
+                Thickness = this.Thickness,
+                Cap = ArcVisual.CapType.Butt,
+                MixColor = TrackColor,
+                StartAngle = 0.0f,
+                SweepAngle = 360.0f
+            };
+            this.AddVisual(TrackVisualName, trackVisual);
+
+            progressVisual = new ArcVisual
+            {
+                SuppressUpdateVisual = true,
+                Size = new Size(this.Size.Width - sliderPadding, this.Size.Height - sliderPadding),
+                SizePolicy = VisualTransformPolicyType.Absolute,
+                Thickness = this.Thickness,
+                Cap = ArcVisual.CapType.Butt,
+                MixColor = ProgressColor,
+                StartAngle = 0.0f,
+                SweepAngle = 0.0f
+            };
+            this.AddVisual(ProgressVisualName, progressVisual);
+
+            thumbVisual = new ArcVisual
+            {
+                SuppressUpdateVisual = true,
+                Size = new Size(this.Size.Width + sliderPadding, this.Size.Height + sliderPadding),
+                SizePolicy = VisualTransformPolicyType.Absolute,
+                Thickness = this.ThumbSize.Width,
+                Cap = ArcVisual.CapType.Round,
+                MixColor = this.ThumbColor,
+                StartAngle = 0.0f,
+                SweepAngle = 0.0f
+            };
+            this.AddVisual(ThumbVisualName,  thumbVisual);
+
+            HandleProgressVisualVisibility();
+
+            UpdateProgressVisualSweepAngle();
+        }
+
+        private void HandleProgressVisualVisibility()
+        {
+            if (isEnabled)
+            {
+                progressVisual.Opacity = 1.0f;
+            }
+            else if (!isEnabled)
+            {
+                progressVisual.Opacity = 0.6f;
+            }
+        }
+
+        private void UpdateVisualThickness(float thickness)
+        {
+            if (trackVisual == null)
+            {
+                return;
+            }
+
+            trackVisual.Thickness = thickness;
+            progressVisual.Thickness = thickness;
+
+            trackVisual.UpdateVisual(true);
+            progressVisual.UpdateVisual(true);
+        }
+
+        private void UpdateProgressVisualSweepAngle()
+        {
+            float progressRatio = (float)(currentValue - minValue) / (float)(maxValue - minValue);
+            float progressWidth = 360.0f * progressRatio; // Circle
+
+            progressVisual.SweepAngle = progressWidth;
+            thumbVisual.StartAngle = progressWidth;
+
+            if (!sweepAngleAnimation)
+            {
+                progressVisual.UpdateVisual(true);
+                thumbVisual.UpdateVisual(true);
+            }
+        }
+
+        private void UpdateAnimation()
+        {
+            // TODO : Currently not sure which effect is needed.
+            AlphaFunction.BuiltinFunctions builtinAlphaFunction = AlphaFunction.BuiltinFunctions.EaseOut;
+
+            if (sweepAngleAnimation)
```

Comment:
`if` condition is duplicated.
```
sweepAngleAnimation?.Stop();
```
 is equivalent to the following code.
```
if (sweepAngleAnimation != null)
   sweepAngleAnimation.Stop();
```
You can remove `if` condition.

Reply:
```
if (animationInstance)
```
This checks whether the Animation instance is a value handle or not (in DALi).
Animation class inherits BaseHandle which is just a handle pointing to an object.
The handle may point to null object sometimes, so the BaseHandle class provides operators to check nullability of the pointing object and the handle itself.

```
/// (In BaseHandle.cs)
public static bool operator true(BaseHandle handle)
{
    // if the handle is null, return false
    if (BaseHandle.ReferenceEquals(handle, null))
    {
        return true;
    }
    // Return whether the handle has a valid reference body or not.
    return handle.HasBody();
}
```


So I think the good code for here is,
```
if (sweepAngleAnimation)
{
   sweepAngleAnimation.Stop();
}
```

Reply:
Ah, i didn't know that. thank you for the notice.